### PR TITLE
Many fixes to py3 support and django 1.7 compatybility

### LINKED ISSUE #327: Move to pytest-django
Refs #325.

### PR DESCRIPTION
**Pull request by galuszkak**
Also adds pytest-django runner and fixes #327 and #318 .

### CODE REVIEW

**Comment by galuszkak** [comment]
I've used updated version of this PR by @montiniz  (Thx!)

**Comment by galuszkak** [comment]
@andrewsmedina @pydanny all tests are passing, it's ready to review and merge :).

File: djadmin2/core.py [hunk 1]
```diff
@@ -122,7 +123,8 @@ def autodiscover(self):
             try:
                 import_module("%s.admin2" % app_name)
             except ImportError as e:
-                if str(e) == "No module named admin2":
+
```
Comment (andrewsmedina) [inline djadmin2/core.py]: can you remove this empty line?

File: djadmin2/views.py [hunk 2]
```diff
@@ -2,9 +2,11 @@
 from __future__ import division, absolute_import, unicode_literals
 
 import operator
+from functools import reduce
```
Comment (galuszkak) [inline djadmin2/views.py]: Actually it's used there, see views.py line 163:
https://github.com/galuszkak/django-admin2/blob/tox_tests/djadmin2/views.py#L163

File: djadmin2/views.py [hunk 3]
```diff
@@ -19,17 +21,14 @@
 from django.utils.text import capfirst
 from django.utils.translation import ugettext_lazy
 from django.views import generic
-
 import extra_views
 
 
 from . import permissions, utils
 from .forms import AdminAuthenticationForm
-from .models import LogEntry
```
Comment (galuszkak) [inline djadmin2/views.py]: Yes, there is, I don't fully remember but this was related to new way how models are initialized in Django 1.7.

File: djadmin2/filters.py [hunk 4]
```diff
@@ -2,18 +2,22 @@
 from __future__ import division, absolute_import, unicode_literals
 
 import collections
+import six
```
Comment (galuszkak) [inline djadmin2/filters.py]: Right, I will fix that.

File: djadmin2/tests/test_types.py [hunk 5]
```diff
@@ -40,7 +40,7 @@ def test_attributes(self):
             self.immutable_admin.d
 
 
-class Thing(models.Model):
+class BigThing(models.Model):
```
Comment (galuszkak) [inline djadmin2/tests/test_types.py]: AFAR I believe there was clash of names and this was raising some errors for django.

File: djadmin2/tests/test_core.py [hunk 6]
```diff
@@ -9,7 +9,7 @@
 from ..core import Admin2
 
 
-class Thing(models.Model):
+class SmallThing(models.Model):
```
Comment (galuszkak) [inline djadmin2/tests/test_core.py]: AFAR I believe there was clash of names and this was raising some errors for django.

File: setup.py [hunk 7]
```diff
@@ -105,16 +125,21 @@ def remove_screenshots(text):
...
+        'django-extra-views>=0.6.5',
+        'django-braces>=1.3.0',
+        'djangorestframework<=2.4.4',
+        'django-floppyforms<=1.2',
+        'django-filter>=0.7',
+        'django-crispy-forms>=1.3.2',
+        'pytz==2014.7'
         ],
+    extras_require={
+        'testing': ['pytest', 'pytest-django', 'pytest-ipdb'],
```
Comment (galuszkak) [inline setup.py]: @andrewsmedina because this fixes #327 and #318

**Comment by galuszkak** [comment]
@andrewsmedina thanks for review, I fixed things you mention.

> merge example + example2
> use py.test

Actually this was done on purpose here. Because some months ago you can see, that I move all tests to tox here. See first comment in this PR because I was fixing this issue: #327 and #318 which as You can see are talking about test runner and using py.test. First I focues moving to py.test to resolve test structure: and this was fine actually to point that I relaised that whole tests won't work when I fix problems that are runner encountering. 
@pydanny agreed that this PR will only be accepted if all tests are passed as for fresh start for next PR.

> whitespace cleanup
> follow pep8 rules

Actually this wasn't on purpose it was just in term of always leave code in better shape than you saw it before. ;)

To be honest, I don't think I can re do this PR in small PR, becuase it was min 1-2 days of work that I did in my spare time, and I don't have now so much free time to get this from 0 point to fully working again... It's actually really big and I know that this bad idea to have so big PR.

**Comment by andrewsmedina** [comment]
@galuszkak thank you!

**Pull request merged**